### PR TITLE
sensors: sensor_preflight_imu -> sensors_status_imu

### PR DESCRIPTION
**Pull request by dagar**
@priseborough here's a quick PR generalizing the preflight imu consistency check to run continuously with the data available per sensor.

 - preflight inconsistencies checks now run always
 - keep inconsistency per sensor

Screen Shot 2020-09-03 at 8 48 12 PM

### CODE REVIEW

File: Tools/ecl_ekf/plotting/pdf_report.py [hunk 1]
```diff
@@ -61,10 +61,10 @@ def create_pdf_report(ulog: ULog, output_plot_filename: str) -> None:
     with PdfPages(output_plot_filename) as pdf_pages:
 
         # plot IMU consistency data
-        if ('accel_inconsistency_m_s_s' in sensor_preflight_imu.keys()) and (
-                'gyro_inconsistency_rad_s' in sensor_preflight_imu.keys()):
+        if ('accel_inconsistency_m_s_s' in sensors_imu_status.keys()) and (
+                'gyro_inconsistency_rad_s' in sensors_imu_status.keys()):
             data_plot = TimeSeriesPlot(
-                sensor_preflight_imu, [['accel_inconsistency_m_s_s'], ['gyro_inconsistency_rad_s']],
+                sensors_imu_status, [['accel_inconsistency_m_s_s'], ['gyro_inconsistency_rad_s']],
```
Comment (dagar) [inline Tools/ecl_ekf/plotting/pdf_report.py:67]: TODO: plot all

**Comment by dagar** [comment]
I've added the sensors module data validator state to `sensors_imu_status` as a health field. This could be useful to check in the ekf2_selector before considering switching to a different instance. The sensor will only be healthy if it hasn't timed out, no stale data, etc.

Screen Shot 2020-09-03 at 10 56 37 PM

**Comment by priseborough** [comment]
@dagar Were you able to run process_logdata_ekf.py on the log file generated after application of these changes? I tried but was unable to do so following the instructions here https://docs.px4.io/master/en/advanced_config/tuning_the_ecl_ekf.html with the following error produced:

  File "process_logdata_ekf.py", line 39
    check_table_filename: str, master_status: str, check_status: Dict[str, str],
                        ^
SyntaxError: invalid syntax

The same error is generated when using master, so if you confirm the same behaviour at your end, i will raise a bug report.

**Review by priseborough — APPROVED** [review]
SITL log data looks OK. We can revisit the LPF coefficient during testing of IMU failure cases after rebasing https://github.com/PX4/Firmware/pull/14650

**Comment by dagar** [comment]
> @dagar Were you able to run process_logdata_ekf.py on the log file generated after application of these changes? I tried but was unable to do so following the instructions here https://docs.px4.io/master/en/advanced_config/tuning_the_ecl_ekf.html with the following error produced:
> 
> File "process_logdata_ekf.py", line 39
> check_table_filename: str, master_status: str, check_status: Dict[str, str],
> ^
> SyntaxError: invalid syntax
> 
> The same error is generated when using master, so if you confirm the same behaviour at your end, i will raise a bug report.

I'll take a look. I was just starting to take a look at extending it for https://github.com/PX4/Firmware/pull/14650#issuecomment-687356690 to process all instances and possibly stop aborting if there's no in air time.

**Comment by mhkabir** [comment]
How about just calling it `imu_status`?

**Comment by dagar** [comment]
> How about just calling it `imu_status`?

It's not a very good overall status (at least not yet) and I wanted to capture it was coming from the `sensors` hub. 

That being said, I think we're overdue for a naming convention discussion. In particular the `vehicle_` prefix is getting a little exhausting.

```
vehicle_acceleration              0    1  1   32 /obj/vehicle_acceleration0
vehicle_air_data                  0    8  1   40 /obj/vehicle_air_data0
vehicle_angular_acceleration      0    1  1   32 /obj/vehicle_angular_acceleration0
vehicle_angular_velocity          0    5  1   32 /obj/vehicle_angular_velocity0
vehicle_attitude                  0    6  1   56 /obj/vehicle_attitude0
vehicle_attitude_setpoint         0    4  1   64 /obj/vehicle_attitude_setpoint0
vehicle_control_mode              0   13  1   32 /obj/vehicle_control_mode0
vehicle_global_position           0    6  1   64 /obj/vehicle_global_position0
vehicle_gps_position              0    5  1  104 /obj/vehicle_gps_position0
vehicle_imu                       0    3  1   56 /obj/vehicle_imu0
vehicle_imu                       1    3  1   56 /obj/vehicle_imu1
vehicle_imu                       2    3  1   56 /obj/vehicle_imu2
vehicle_imu_status                0    1  1   56 /obj/vehicle_imu_status0
vehicle_imu_status                1    4  1   56 /obj/vehicle_imu_status1
vehicle_imu_status                2    4  1   56 /obj/vehicle_imu_status2
vehicle_land_detected             0    9  1   24 /obj/vehicle_land_detected0
vehicle_local_position            0   12  1  168 /obj/vehicle_local_position0
vehicle_magnetometer              0    2  1   32 /obj/vehicle_magnetometer0
vehicle_odometry                  0    1  1  256 /obj/vehicle_odometry0
vehicle_rates_setpoint            0    4  1   32 /obj/vehicle_rates_setpoint0
vehicle_status                    0   21  1   56 /obj/vehicle_status0
vehicle_status_flags              0    0  1   40 /obj/vehicle_status_flags0
vehicle_visual_odometry_aligned   0    0  1  256 /obj/vehicle_visual_odometry_aligned0
```

``` Console
sensors/
├── data_validator
│   └── tests
├── vehicle_acceleration
├── vehicle_air_data
├── vehicle_angular_velocity
├── vehicle_imu
└── vehicle_magnetometer
    └── mag_compensation
        └── python
```

To be continued...

**Comment by dagar** [comment]
> @dagar Were you able to run process_logdata_ekf.py on the log file generated after application of these changes? I tried but was unable to do so following the instructions here https://docs.px4.io/master/en/advanced_config/tuning_the_ecl_ekf.html with the following error produced:
> 
> File "process_logdata_ekf.py", line 39
> check_table_filename: str, master_status: str, check_status: Dict[str, str],
> ^
> SyntaxError: invalid syntax
> 
> The same error is generated when using master, so if you confirm the same behaviour at your end, i will raise a bug report.

I have it working now after a few minor changes related for the new `estimator_states` introduced in the supporting PR https://github.com/PX4/Firmware/pull/15657. I wasn't getting the same failure as you on either Ubuntu 18.04 or a Mac.

Do you know if you were using python2 or 3? If it's still failing for you let's follow up in an issue.

**Pull request merged**